src/core/checkpoint_manager.py

```
@@ -75,7 +75,22 @@ def find_checkpoint_for_file(self, output_file: str) -> Optional[Dict]:
     def get_or_create_output_file(self, country_code: str, resume: bool = True) -> Tuple[Path, Optional[Dict]]:
         """Get existing output file or create new one"""
         if resume:
-            # Look for existing checkpoints for this country
+            # First check for existing checkpoint file
+            checkpoint_file = self.checkpoint_dir / f"checkpoint_{country_code.lower()}_all_parallel.json"
+            if checkpoint_file.exists():
+                with open(checkpoint_file, 'r') as f:
```

Ruff (0.12.2)

Unnecessary mode argument

Remove mode argument

(UP015)